Client/Assets/ContextMenu.js

```diff
@@ -74,10 +74,37 @@ addEventListener("touchstart", function (event) {
   var data = {};
   var element = event.target;
 
+  element.addEventListener("touchend", function (event) {
+    cancel();
+
+    // If we're showing the context menu, prevent the page from handling the click event.
+    if (touchHandled) {
+      touchHandled = false;
+      event.preventDefault();
+    }
+  }, true);
+
+  element.addEventListener("touchmove", function (event) {
+    if (longPressTimeout) {
+      var { screenX, screenY } = event.touches[0];
+
+      // Cancel the context menu if finger has moved beyond the maximum allowed distance.
+      if (Math.abs(touchDownX - screenX) > MAX_RADIUS || Math.abs(touchDownY - screenY) > MAX_RADIUS) {
+        cancel();
+      }
+    }
+  });
+
   do {
     if (!data.link && element.localName === "a") {
       data.link = element.href;
 
+      // Don't show the context menu if this is a script link
+      if (element.getAttribute("href") === "#") {
+        data = {};
+        break;
+      }
```

Comment:
There's actually no such thing as a "script link" -- it's just a regular `<a>` element with an `onclick` event attached to it. For example, the [demo on my people page](https://people.mozilla.org/~bnicholson/test/touchend.html) uses **#foo** rather than **#**, and it works the same. Or the page could choose to attach an `onclick` listener to a link that doesn't have an anchor at all.

http://foo.com/#, http://foo.com/#bar, etc. are all valid URLs, so we probably shouldn't treat them any differently from normal links. That is, I'd still show the context menu for such elements (as long as the user actually does a long press).